src/gui/qgsmaptooladvanceddigitizing.h

```diff
@@ -107,6 +107,9 @@ class GUI_EXPORT QgsMapToolAdvancedDigitizing : public QgsMapToolEdit
      */
     void setAutoSnapEnabled( bool enabled ) { mAutoSnapEnabled = enabled; }
 
+
+    QgsAdvancedDigitizingDockWidget *mCadDockWidget = nullptr;
```

Comment:
Wouldn't there be a better approach than making this protected? Such as forwarding undo/redo to the dock the same way we do with mouse events?

Reply:
Hmm, actually there's already this :
```
bool QgsAdvancedDigitizingDockWidget::canvasKeyPressEventFilter( QKeyEvent *e )
{
  ...
  switch ( e->key() )
  {
    case Qt::Key_Backspace:
    case Qt::Key_Delete:
    {
      removePreviousPoint();
      releaseLocks( false );
      break;
    }
    ...
}
```
Is this what you were referring to ? Anyway, it seems these lines have no effect in the case of QgsMapToolCapture, maybe because of `e->ignore();` in ` QgsMapToolCapture::keyPressEvent` ?

To me it seems right that the remove point behaviour is done by the map tool rather than by the CadDockWidget, as the delete key could potentially work quite differently from one map tool to another. Also it's consistent with the how it's done for adding points  (e.g. `mCadDockWidget->addPoint( e->mapPoint() );` in `QgsMapToolAdvancedDigitizing::canvasReleaseEvent`).

Let me know what you think !

Reply:
@3nids confirmed that this looks good, thanks!